src/olympia/lib/tests/test_git.py

```diff
@@ -376,6 +376,40 @@ def test_extract_and_commit_source_from_version_no_dotgit_clash(settings):
         'extracted/manifest.json', 'extracted/.git.b236f599/config'}
 
 
+@pytest.mark.django_db
+def test_extract_and_commit_source_from_version_rename_dotgit_files(settings):
+    addon = addon_factory(
+        file_kw={'filename': 'webextension_no_id.xpi'},
+        version_kw={'version': '0.1'})
+
+    # Generate source file
+    source = temp.NamedTemporaryFile(suffix='.zip', dir=settings.TMP_PATH)
+    with zipfile.ZipFile(source, 'w') as zip_file:
+        zip_file.writestr('manifest.json', '{}')
+        zip_file.writestr('.gitattributes', '')
+        zip_file.writestr('.gitignore', '')
```

Comment:
Is there a test to make sure nested `.gitignore` files are also renamed? I can see that they are but this is something we should have a regression test for since it's super important.

Reply:
I don't think we test that indeed, all files to rename could potentially be nested. Could you file an issue please?

Reply:
Sure. Filed https://github.com/mozilla/addons-server/issues/11370